test/test_reader.py

```diff
@@ -640,3 +640,81 @@ def test_heuristic_for_3D_works_as_expected(dset):
     fh.seek(0)
     reader = create_reader(fh, dimensionality=None)
     assert reader.dimensionality == 3
+
+
+@given(dataset_3d(valid_line_catalog=True))
+@settings(
+    max_examples=10,
+    suppress_health_check=(HealthCheck.too_slow,),
+    deadline=None,
+    phases=(Phase.explicit, Phase.reuse, Phase.generate),
+)
+def test_num_inlines_is_correct(dset):
+    fh = io.BytesIO()
+    write_segy(fh, dset)
+    fh.seek(0)
+    reader = create_reader(fh)
+    assert reader.num_inlines() == dset.num_traces()
+
+
+@given(dataset_3d(valid_line_catalog=True))
+@settings(
+    max_examples=10,
+    suppress_health_check=(HealthCheck.too_slow,),
+    deadline=None,
+    phases=(Phase.explicit, Phase.reuse, Phase.generate),
+)
+def test_num_xlines_is_correct(dset):
+    fh = io.BytesIO()
+    write_segy(fh, dset)
+    fh.seek(0)
+    reader = create_reader(fh)
+    assert reader.num_xlines() == dset.num_traces()
+
+
+@given(dataset_3d(valid_line_catalog=True))
+@settings(
+    max_examples=10,
+    suppress_health_check=(HealthCheck.too_slow,),
+    deadline=None,
+    phases=(Phase.explicit, Phase.reuse, Phase.generate),
+)
+def test_inline_xline_numbers_is_correct(dset):
+    fh = io.BytesIO()
+    write_segy(fh, dset)
+    fh.seek(0)
+    reader = create_reader(fh)
+    assert sorted(reader.inline_xline_numbers()) == sorted((idx, idx) for idx in range(dset.num_traces()))
```

Comment:
Seems to me that this assertion would only hold if the only traces were along the diagonal (0, 0), (1, 1), (2, 2),...(n, n) of the inline, crossline grid. Aha! I think I see now that that is deliberately what you're creating, because it has these interesting qualities. Maybe we should name the strategy as such then - diagonal_dataset_3d or something...

Reply:
Right, the datasets I'm creating are pretty "artificial", perhaps too much so. A renaming is probably the right thing to do.